sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -294,7 +294,7 @@ private void createRunnerAndConsumersForPTransformRecursively(
 
         createRunnerAndConsumersForPTransformRecursively(
             beamFnStateClient,
-            queueingClient,
+            metricsClient,
             entry.getKey(),
             entry.getValue(),
             request::getInstructionId,
```

Comment:
A one line change is needed to fix the test failing right now, I'll take care of this tomorrow morning.

      try (Closeable closeTracker = stateTracker.activate()) {

// becomes
 try (Closeable closeTracker = MetricContainerStepMapEnvironment.activate()) {

needs to use the Metric environment. This is because Its possible the ElementCountFnDataReceiver which comes out of the MetricPCollectionConsumerRegistry can be called from finish() functions which occur on this thread. So we have to use the pattern I introduced in this PR, in this thread as well. Which I originally thought was unnecessary

See the finish function added in CombineRunners.java:
https://github.com/apache/beam/blob/5b1700db834cf949547a20a30f3ee1270710585a/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/CombineRunners.java#L176

https://github.com/apache/beam/blob/5b1700db834cf949547a20a30f3ee1270710585a/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/CombineRunners.java#L103